homework/hm11_bank_card_operations.py

```diff
@@ -0,0 +1,186 @@
+-Use CVV, PIN, Name, Surname , end date, card_num as initial params.
+-Class should have in addition to common logic some class attributes,
+as minimum one class method and as minimum one staticmethod,
+two or more getters/setters, __str_ magic method.
+-Do not forget about block ""if __name__ == '__main__':""
+and code there to check your class written logic.
+"""
+
+TERMINAL_SEPARATOR = '-' * 42
+
+
+class BankCard:
+    bank_name = "StereoBank"
+    currency = "USD"
+
+    def __init__(self, name, surname, card_number, pin, cvv, end_date):
+        self.__card_statement = 0.00
+        self.__is_blocked = False
+        self.__name = name
+        self.__surname = surname
+        self.__card_number = card_number
+        self.__pin = pin
+        self.__cvv = cvv
+        self.__end_date = end_date
+
+    @classmethod
+    def get_bank_info(cls):
+        return f'Bank: {cls.bank_name}, Currency: {cls.currency}'
+
+    @classmethod
+    def get_currency(cls):
+        return cls.currency
+
+    def open_account(self):
+        self.__card_statement = 0.00
+
+    def check_balance(self):
+        return self.__card_statement
+
+    @staticmethod
+    def validate_card_number(card_number):
+        if len(str(card_number)) != 16:
+            return False
+        return True
+
+    def top_up_balance(self, amount):
+        self.__card_statement = self.__card_statement + amount
+        return amount
+
+    def top_down_balance(self, amount):
+        if self.__card_statement - amount < 0:
+            raise (ValueError
+                   ('There are not enough funds to perform this operation.'))
+        self.__card_statement = self.__card_statement - amount
+        return amount
+
+    def transfer_money(self, to_card, amount):
+        if self.__card_statement - amount < 0:
+            raise (ValueError
+                   ('There are not enough funds to perform this operation.'))
+        self.__card_statement -= amount
+        to_card.top_up_balance(amount)
+        return amount
+
+    def block_card(self):
+        self.__is_blocked = True
+
+    def unblock_card(self):
+        self.__is_blocked = False
+
+    def is_blocked(self):
+        return self.__is_blocked
+
+    def close_account(self, to_card=None):
+        if self.__card_statement != 0:
+            if to_card:
+                to_card.top_up_balance(self.__card_statement)
+                self.__card_statement = 0.00
+                return (f'The account closed successfully. '
+                        f'Remaining balance transferred to '
+                        f'{to_card.__name} {to_card.__surname}.')
+            else:
+                return ("The account cannot be closed "
+                        "as it has a positive balance. "
+                        "Please provide account to transfer funds.")
+        else:
+            self.__card_statement = 0.00
+            return "The account closed successfully."
+
+    def get_name(self):
```

Comment:
> Plus do not forget to write setters and getters with property decorator

but this is not a prerequisite for the task.

Reply:
I removed some getters and setters.
Please, look at the ones that remain. In your opinion, are there any others that need to be removed?
set for name/surname I left it because the user can change the first and last name. for example in the case of marriage.